python/ray/data/preprocessors/encoder.py

```diff
@@ -536,7 +536,11 @@ def update_counter(element):
         return Counter(col.value_counts(dropna=False).to_dict())
 
     def get_pd_value_counts(df: pd.DataFrame) -> List[Dict[str, Counter]]:
-        result = [{col: get_pd_value_counts_per_column(df[col]) for col in columns}]
+        for col in columns:
+            if col in df.columns.tolist():
```

Comment:
This `df.columns.tolist()` is potentially expensive, since it will convert the internal Pandas representation of the column name collection to a Python list. Could you pull this outside of the for loop?